homeassistant/components/daikin/climate.py

```diff
@@ -227,19 +239,51 @@ def preset_mode(self):
             == HA_PRESET_TO_DAIKIN[PRESET_AWAY]
         ):
             return PRESET_AWAY
+        if (
+            HA_PRESET_TO_DAIKIN[PRESET_BOOST]
+            in self._api.device.represent(DAIKIN_ATTR_ADVANCED)[1]
+        ):
+            return PRESET_BOOST
+        if (
+            HA_PRESET_TO_DAIKIN[PRESET_ECO]
+            in self._api.device.represent(DAIKIN_ATTR_ADVANCED)[1]
+        ):
+            return PRESET_ECO
         return PRESET_NONE
 
     async def async_set_preset_mode(self, preset_mode):
         """Set preset mode."""
         if preset_mode == PRESET_AWAY:
             await self._api.device.set_holiday(ATTR_STATE_ON)
+        elif preset_mode == PRESET_BOOST:
```

Comment:
I only have a device that does support these modes so I can't test how a device that doesn't support them reacts. The pydaikin library itself only raises an exception if the device responds with 403 Forbidden, other errors are apparently ignored. I'd assume the device would respond with 404 if no advanced modes are supported and with 200 OK and an error code in the response if the desired mode isn't supported (this is what happens with my device in other cases).

Since the proprietary controller protocol isn't documented it's quite difficult to reason about what happens when, without having access to a large amount of test devices.